http/src/main/resources/init-resources/init-actions.sh

```diff
@@ -135,6 +136,30 @@ function apply_start_user_script() {
 # .. after jupyter notebook start
 # END
 STEP_TIMINGS=($(date +%s))
+
+
+## Installs Google Cloud Ops Agent that is now required for Datapoc 2.2.X ###
+# See https://github.com/GoogleCloudDataproc/initialization-actions/tree/master/opsagent
+# Installs the Google Cloud Ops Agent on each node in the cluster.
+# It also provides an override to the built-in logging config to set empty
+# receivers i.e. not collect any logs.
+# If you need to collect syslogs, you can use the other script in this directory,
+# opsagent.sh which uses the built-in configuration of Ops Agent.
+# See https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/configuration#default.
+#
+# Detect dataproc image version from its various names
+if (! test -v DATAPROC_IMAGE_VERSION) && test -v DATAPROC_VERSION; then
+  DATAPROC_IMAGE_VERSION="${DATAPROC_VERSION}"
+fi
+
+if [[ $(echo "${DATAPROC_IMAGE_VERSION} < 2.2" | bc -l) == 1  ]]; then
+  echo "This Dataproc cluster node runs image version ${DATAPROC_IMAGE_VERSION} with pre-installed legacy monitoring agent. Skipping Ops Agent installation."
+  exit 0
+fi
```

Comment:
Won't this never be hit because we detect the legacy AOU version and use the special legacy aou dataproc init script instead in that case?

Reply:
Yeah, I can remove, see my latest commit, I decided to go with a completely separate init file for the legacy AOU case, I can remove that :)